File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        res = 0
        before = 0
        after = 0
        for num in nums:
            if num == 1:
                after += 1
            else:
                res = max(res, before + after)
                before = after
                after = 0
        res = max(res, before + after)
        return res if res != len(nums) else res - 1

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int res = 0;
        int before = 0;
        int after = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            if(nums[i] == 1)
            {
                after++;
            }
            else
            {
                res = max(res, before + after);
                before = after;
                after = 0;
            }
        }
        res = max(res, before + after);
        return res != nums.size() ? res : res - 1;
    }
};